addons/mail/models/mail_compose_message.py

```diff
@@ -643,17 +666,101 @@ def _gc_lost_attachments(self):
             ('write_date', '<', limit_date)]
         ).unlink()
 
+    @api.autovacuum
+    def _gc_not_scheduled(self):
+        """ Garbage collect composers that are not scheduled (drafts that were not scheduled nor
+        sent, and sent composers) """
+        limit_date = fields.Datetime.subtract(fields.Datetime.now(), days=1)
+        self.search([('state', '!=', 'scheduled'), ('write_date', '<', limit_date)]).unlink()
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        composers = super().create(vals_list)
+        # link attachments to composer (they were created with res_id=0)
+        for composer in composers:
+            if attachments := composer.attachment_ids:
```

Comment:
Note: scheduling will probably have an impact in ``_process_attachments_for_post`` as we take attachments linked to composer model (ok, does not change) and created by self._uid, so we should be sure everything is done with the right uid.